graylog2-server/src/main/java/org/graylog2/outputs/ElasticSearchOutput.java

```diff
@@ -112,6 +112,7 @@ public void writeMessageEntries(List<Map.Entry<IndexSet, Message>> messageList)
         }
         failures.mark(failedMessageIds.size());
 
+        // TODO why doesn't this exclude the failedMessageIds?
```

Comment:
I think we don't exclude the failed IDs because we don't know if we can retry the failed messages. If we have a message that couldn't be written to ES due to a data type error and we don't commit its offset, then it will be retried over and over again. Okay, in practice it will most probably not because that would require the message to be the last one.

But then it also doesn't make sense to exclude the offset either because if the faulty message is in the middle of a batch and we commit the last offset, it's gone anyway.

Reply:
fair enough. should I add that to the comment?